src/web-fls-quiz/Services/QuestionService.cs

```diff
@@ -16,9 +16,9 @@ public QuestionService(IDataStorage dataStorage)
             _dataStorage = dataStorage;
         }
         
-        public Question GetRandom(int[] excludedQuestionIds)
+        public Question GetRandom(int[] excludedQuestionIds, int quizId)
         {
-            var availableIds = Enumerable.Range(1, (int)_dataStorage.GetQuestionsNumber()).Except(excludedQuestionIds).ToArray();
+            var availableIds = _dataStorage.GetQuestionIds(quizId).ToList().Except(excludedQuestionIds).ToArray();
```

Comment:
Logic should accept a quiz name. Client code has no business in dealing with concrete ids.